indigo/analysis/refs/base.py

```diff
@@ -91,44 +92,45 @@ def setup(self, root):
         super().setup(root)
 
     def setup_subtypes(self):
-        self.subtypes = [s for s in Subtype.objects.all()]
-        self.subtype_names = [s.name for s in self.subtypes]
-        self.subtype_abbreviations = [s.abbreviation for s in self.subtypes]
-
-        self.subtypes_string = '|'.join([re.escape(s) for s in self.subtype_names + self.subtype_abbreviations])
+        self.full_subtypes = Subtype.objects.all()
+        self.subtypes = [s.name for s in self.full_subtypes] + \
+                        [s.abbreviation for s in self.full_subtypes]
 
     def setup_candidate_xpath(self):
-        xpath_contains = " or ".join([f"contains(translate(., '{subtype.upper()}', '{subtype.lower()}'), "
-                                      f"'{subtype.lower()}')"
-                                      for subtype in self.subtype_names + self.subtype_abbreviations])
+        xpath_contains = " or ".join(
+            [f"contains(translate(., '{s.upper()}', '{s.lower()}'), "
+             f"'{s.lower()}')"
+             for s in self.subtypes])
         self.candidate_xpath = f".//text()[({xpath_contains}) and not(ancestor::a:ref)]"
 
     def setup_pattern_re(self):
-        # TODO: disregard e.g. "6 May" in "GN 34 of 6 May 2020", but catch reference
+        subtypes_for_re = '|'.join([re.escape(s) for s in self.subtypes])
         self.pattern_re = re.compile(
             fr'''
                 (?P<ref>
-                    (?P<subtype>{self.subtypes_string})\s*
+                    (?P<subtype>{subtypes_for_re})\s*
                     (No\.?\s*)?
                     (?P<num>\d+)
-                    (\s+of\s+|/)
-                    (?P<year>\d{{4}})
+                    .{{0,60}}(\s|/)
```

Comment:
This makes me a bit nervous, it's pretty broad.

Reply:
Neil had an interesting idea which maybe we need to start exploring: start with a broad regex, and then filter down edge cases with supplemental regexes. His argument is it helps to keep the regexes simple and understandable.

Reply:
eg. this would be a false positive: `as published in GN No 102. In section 4378 of the Act...`

This stuff is tricky.